cf/commands/domain/domains_test.go

```diff
@@ -1,137 +1,379 @@
+		repoLocator := deps.RepoLocator.SetRoutingApiRepository(routingApiRepo)
+
+		domainRepo = &fakeapi.FakeDomainRepository{}
+		repoLocator = repoLocator.SetDomainRepository(domainRepo)
 
-	runCommand := func(args ...string) bool {
-		return testcmd.RunCliCommand("domains", args, requirementsFactory, updateCommandDependency, false)
-	}
+		deps = command_registry.Dependency{
+			Ui:          ui,
+			Config:      configRepo,
+			RepoLocator: repoLocator,
+		}
 
-	Describe("requirements", func() {
-		It("fails when an org is not targeted", func() {
-			requirementsFactory.LoginSuccess = true
+		cmd = domain.ListDomains{}
+		cmd.SetDependency(deps, false)
 
-			Expect(runCommand()).To(BeFalse())
+		flagContext = flags.NewFlagContext(cmd.MetaData().Flags)
+
+		factory = &fakerequirements.FakeFactory{}
+
+		loginRequirement = &passingRequirement{Name: "LoginRequirement"}
+		factory.NewLoginRequirementReturns(loginRequirement)
+
+		routingApiRequirement = &passingRequirement{Name: "RoutingApiRequirement"}
+		factory.NewRoutingAPIRequirementReturns(routingApiRequirement)
+
+		targetedOrgRequirement = &fakerequirements.FakeTargetedOrgRequirement{}
+		factory.NewTargetedOrgRequirementReturns(targetedOrgRequirement)
+
+		callBackFunc = func(orgGuid string,
+			cb func(models.DomainFields) bool) error {
+			for _, field := range domainFields {
+				if !cb(field) {
+					break
+				}
+			}
+			return nil
+		}
+
+	})
+
+	Describe("Requirements", func() {
+		Context("when provided one arg", func() {
+			BeforeEach(func() {
+				flagContext.Parse("arg-1")
+			})
+
+			It("fails with usage", func() {
+				Expect(func() { cmd.Requirements(factory, flagContext) }).To(Panic())
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"Incorrect Usage. No argument required"},
+					[]string{"NAME"},
+					[]string{"USAGE"},
+				))
+			})
 		})
 
-		It("fails when not logged in", func() {
-			requirementsFactory.TargetedOrgSuccess = true
+		Context("when provided no arguments", func() {
+			BeforeEach(func() {
+				flagContext.Parse()
+			})
+
+			It("does not fail with usage", func() {
+				Expect(ui.Outputs).NotTo(ContainSubstrings(
+					[]string{"Incorrect Usage. No argument required"},
+					[]string{"NAME"},
+					[]string{"USAGE"},
+				))
+			})
+
+			It("returns a LoginRequirement", func() {
+				actualRequirements, err := cmd.Requirements(factory, flagContext)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(factory.NewLoginRequirementCallCount()).To(Equal(1))
+				Expect(actualRequirements).To(ContainElement(loginRequirement))
+			})
+
+			It("returns a TargetedOrgRequirement", func() {
+				actualRequirements, err := cmd.Requirements(factory, flagContext)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(factory.NewTargetedOrgRequirementCallCount()).To(Equal(1))
+				Expect(actualRequirements).To(ContainElement(targetedOrgRequirement))
+			})
 
-			Expect(runCommand()).To(BeFalse())
+			It("does not return a RoutingAPIRequirement", func() {
```

Comment:
This test doesn't make any sense; in no situation would you ever return the RoutingAPIRequirement. This test could not have failed.